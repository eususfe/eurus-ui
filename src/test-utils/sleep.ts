/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
