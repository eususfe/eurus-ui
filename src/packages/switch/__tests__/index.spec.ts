import { mount } from '@vue/test-utils';
import ESwitch from '..';

describe('ESwitch', () => {

  it('ESwitch snapshot', () => {
    const wrapper = mount(ESwitch);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
});
