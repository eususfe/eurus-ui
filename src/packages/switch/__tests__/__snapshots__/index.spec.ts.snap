// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ESwitch > ESwitch snapshot 1`] = `"<label class=\\"eu-switch\\"><input class=\\"eu-switch__input\\" type=\\"checkbox\\" role=\\"switch\\" aria-checked=\\"false\\"><span class=\\"eu-switch__inner\\"></span></label>"`;

exports[`ESwitch > should emit change event 1`] = `"<label class=\\"eu-switch\\"><input class=\\"eu-switch__input\\" type=\\"checkbox\\" role=\\"switch\\" aria-checked=\\"false\\"><span class=\\"eu-switch__inner\\"></span></label>"`;

exports[`ESwitch > should work with \`disabled\` props 1`] = `"<label class=\\"eu-switch eu-switch-disabled\\"><input class=\\"eu-switch__input\\" type=\\"checkbox\\" role=\\"switch\\" aria-checked=\\"false\\" aria-disabled=\\"true\\" disabled=\\"\\"><span class=\\"eu-switch__inner\\"></span></label>"`;
